ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/modules/sessions/services/SessionsValidators.kt

```diff
@@ -1,29 +1,30 @@
 package com.izivia.ocpi.toolkit.modules.sessions.services
 
-import com.izivia.ocpi.toolkit.common.validation.*
+import com.izivia.ocpi.toolkit.common.validation.hasMaxLengthOf
+import com.izivia.ocpi.toolkit.common.validation.isCountryCode
+import com.izivia.ocpi.toolkit.common.validation.isCurrencyCode
+import com.izivia.ocpi.toolkit.common.validation.isPrintableAscii
 import com.izivia.ocpi.toolkit.modules.cdr.domain.*
 import com.izivia.ocpi.toolkit.modules.sessions.domain.*
 import com.izivia.ocpi.toolkit.modules.types.Price
 import com.izivia.ocpi.toolkit.modules.types.PricePartial
 import com.izivia.ocpi.toolkit.modules.types.toPartial
-import org.valiktor.functions.isGreaterThan
 import org.valiktor.functions.isGreaterThanOrEqualTo
 import org.valiktor.validate
 import java.math.BigDecimal
-import java.time.Instant
 
 fun SessionPartial.validate(): SessionPartial = validate(this) {
     validate(SessionPartial::countryCode).isCountryCode(caseSensitive = false, alpha2 = true)
     validate(SessionPartial::partyId).isPrintableAscii().hasMaxLengthOf(3)
     validate(SessionPartial::id).isPrintableAscii().hasMaxLengthOf(36)
-    validate(SessionPartial::startDateTime).isGreaterThan(Instant.now())
-    validate(SessionPartial::endDateTime).isGreaterThan(it.startDateTime ?: Instant.now())
+    // startDateTime nothing to validate
+    // endDateTime nothing to validate
```

Comment:
I suggest to validate that the dates are the current time or in the past.

```suggestion
    validate(SessionPartial::startDateTime).isLessThanOrEqualTo(Instant.now())
    validate(SessionPartial::endDateTime).isLessThanOrEqualTo(it.startDateTime ?: Instant.now())
```

Reply:
Maybe that was the original intent, just went into the wrong direction ... I'm just not entirely sure how far this kind of validation brings in value, as it is relative. If I have a partner with a badly synced time (or timezone), sending me sessions with timestamps one hour into the future, I'll fail validation at one point, and will be perfectly happy with them later ... it's not quite the same as checking for positive value on a number. For the most part, we care that it is a valid date, and the serializer is taking care of that

we can always add additional business logic on the repository side, where we have the full object, to ensure enddate is higher, etc. Because in a Session Patch, you'll never have start and end time together, so it is very unlikely we ever do that check

Reply:
I agree. This is a functional validation that each application can implement in a different way. It shouldn't be on the library.

Reply:
I agree with you, we should not validate start/end times in the lib. Moreover we should never use Instant.now() which is relative to the systeme timezone. The good practice is to pass a clock, which is set up on the right timezone (fixed, not system-dependant), and to call clock.now().